shm/Map.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <random>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+#include "Island.hpp"
```

Comment:
Includes should be sorted alphabetically. Moreover  headers and standard library includes should be in seperate paragraph.